src/bp/core/services/hints/file-based/execute-vars-provider.ts

```diff
@@ -0,0 +1,53 @@
+import { FlowView } from 'common/typings'
+
+import { FileBasedHintProvider, Hint } from '..'
+
+const USER_VARIABLE_REGEX = [
+  /event.state.(temp|session|user|bot).([A-Z0-9_]+?)\s?=/gi,
+  /(temp|session|user|bot).([A-Z0-9_]+?)\s?=/gi
+]
+
+const MIN_VARIABLE_LENGTH = 3
+
+export default class ActionFlowVariablesProvider implements FileBasedHintProvider {
```

Comment:
I feel it a bit odd to implement a hint provider in the core as this is clearly code-editor releated ..

Reply:
Not entirely code-editor related; the hint provider also provides autocompletion for super input & elsewhere. I added a provider so custom code files were also parsed, and not only the actions on the full editor